src/session_impl.cpp

```diff
@@ -1655,6 +1655,9 @@ namespace aux {
 	// session_impl is responsible for deleting 'pack'
 	void session_impl::apply_settings_pack(boost::shared_ptr<settings_pack> pack)
 	{
+#if TORRENT_USE_INVARIANT_CHECKS
+		check_invariant();
+#endif
 		apply_settings_pack_impl(*pack);
```

Comment:
you need to use the macro here, otherwise the invariant check is not checked at the end of the function